src/utils/index.ts

```diff
@@ -617,6 +617,32 @@ export async function shellWithDocker(): Promise<vscode.Terminal> {
   return terminal;
 }
 
+interface WSServerRootFolderData {
+  redirectDotvscode: boolean;
+}
+
+const wsServerRootFolders = new Map<string, WSServerRootFolderData>();
+
+/**
+ * Add uri to the wsServerRootFolders map if eligible
+ */
+export async function addWsServerRootFolderData(uri: vscode.Uri): Promise<void> {
+  if (!schemas.includes(uri.scheme)) {
+    return;
+  }
+  const value: WSServerRootFolderData = {
+    redirectDotvscode: true,
+  };
+  if (isCSPFile(uri)) {
```

Comment:
You should add a check for `!["","/"].includes(uri.path)` because if that's true you have to redirect to the namespace version.

Reply:
I'm not clear what you mean.

Reply:
Sorry, I meant that if the root folder is showing all web apps and not a specific web app, then you need to redirect to the namespace settings.json anyways and there's no need to ping the server.

Reply:
OK, I think I undertand.